### PR TITLE
Security management enhancements:

### PR DESCRIPTION
**Pull request by dgutierr**
UF-325: Add help icons with tips to workbench permissions
UF-329: Add automatic disabling of illogical combinations in Security perspective

NOTE: this PR depends on https://github.com/uberfire/uberfire/pull/432

@manstis Can you verify please?

### CODE REVIEW

File: guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/security/ProjectTreeProvider.java [hunk 1]
```diff
@@ -73,12 +73,22 @@ public void setRootNodePosition(int rootNodePosition) {
...
+        Permission buildPermission = newPermission(BUILD);
+        Permission createPermission = newPermission(CREATE);
+
+        rootNode.addPermission(readPermission, i18n.ProjectActionRead());
+        rootNode.addPermission(updatePermission, i18n.ProjectActionUpdate());
+        rootNode.addPermission(deletePermission, i18n.ProjectActionDelete());
+        rootNode.addPermission(buildPermission, i18n.ProjectActionBuild());
+        rootNode.addPermission(createPermission, i18n.ProjectActionCreate());
+
+        rootNode.addDependencies(readPermission, updatePermission, deletePermission, buildPermission);
```
Comment (manstis) [inline guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/security/ProjectTreeProvider.java]: @dgutierr Perhaps I need to read the UF PR.. but why is there a need to add permissions to ```rootNode``` and also add some of the permissions as dependencies?

Comment (dgutierr) [inline guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/security/ProjectTreeProvider.java]: @manstis  What that lines does is to add the update, delete & build permissions as dependencies of the read permission. Just take a look at the "addDependencies" javadoc,

Comment (manstis) [inline guvnor-project/guvnor-project-client/src/main/java/org/guvnor/common/services/project/client/security/ProjectTreeProvider.java]: @dgutierr Thanks.. read the new JavaDoc in the related UF PR and it all makes sense now.

**Comment by manstis** [comment]
@dgutierr Shouldn't we have tests to check permissions are correctly set-up?

**Comment by dgutierr** [comment]
UF already provides coverage for the permission inter-dependencies.

Anyway, I also think is a good idea to have some coverage for the tree providers. I'll prepare some. Thanks for the remainder...

**Comment by manstis** [comment]
@dgutierr Yes, I assume Uberfire has tests to check the generic framework works correctly; I was meaning, as you say, some specific tests for the different tree providers to verify the default permissions are correctly configured.

**Comment by dgutierr** [comment]
@manstis Tests added

**Comment by manstis** [comment]
Great, thanks @dgutierr 

+1 to merge (when UF PR is ready).

**Pull request merged**